datadriven/src/sgpp/datadriven/algorithm/DBMatOffline.cpp

```diff
@@ -146,7 +147,7 @@ void DBMatOffline::store(const std::string& fileName) {
   }
 
   std::string inter = "," + std::to_string(interactions.size());
-  for (std::vector<size_t> i : interactions) {
+  for (std::set<size_t> i : interactions) {
```

Comment:
I would replace all the range loop variables over non-integral types with references for speed reasons.
```suggestion
  for (const std::set<size_t>& i : interactions) {
```